File: LC74.cpp
class Solution {
public:
    bool searchMatrix(vector<vector<int>>& matrix, int target) {
        
        int low = 0, high = matrix.size()-1;

        while(low <= high){
            int mid = (low+high)/2;

            if(target == matrix[mid][0])
                return true;
            
            if(target > matrix[mid][0])
                low = mid + 1;
            else 
                high = (mid - 1);
        }

        int row = min(low, high) < 0 ? 0 : min(low,high);

        low = 0; high = matrix[0].size()-1;

        while(low <= high){
            int mid = (low+high)/2;

            if(target == matrix[row][mid])
                return true;
            if(target > matrix[row][mid])
                low = mid+1;
            else 
                high = mid-1;
        }

        return false;

    }
};

File: LC74.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:

        l = 0
        h = len(matrix) - 1

        while l <= h:
            m = (l+h)//2

            if target == matrix[m][0]:
                return True
            
            if target < matrix[m][0]:
                h = m - 1
            else:
                l = m + 1
        
        row = min(l,h) if min(l,h) > 0 else 0

        l = 0
        h = len(matrix[row]) - 1

        while l <= h:
            m = (l+h)//2

            if target == matrix[row][m]:
                return True
            
            if target < matrix[row][m]:
                h = m - 1
            else:
                l = m + 1
        
        return False
